### PR TITLE
Fix readjustment of number of slices bug in RSA

### PR DESCRIPTION
**Pull request by Rosejoycrocker**
Fixes readjustment of number of slices bug in RSA (partially addresses #615) and changes output of RSA to `YAXArrays.Dataset` type.

### CODE REVIEW

**Comment by ConnectedSystems** [comment]
Does the switch to YAXArray break the ability to plot results? 
I don't see any compatibility layer in place.

**Comment by Rosejoycrocker** [comment]
> Does the switch to YAXArray break the ability to plot results? I don't see any compatibility layer in place.

There are also changes to `viz.rsa()` in this PR so plotting works

File: ext/AvizExt/viz/viz.jl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 using Makie, DataFrames
 
 using ADRIA: ResultSet, metrics.metric_label, analysis.col_normalize, model_spec
-using NamedDims, AxisKeys
+using NamedDims, AxisKeys, YAXArrays, DimensionalData
```
Comment (ConnectedSystems) [inline ext/AvizExt/viz/viz.jl:4]: Is the import of DimensionalData simply to have the `At` method?

If so, I'd prefer it be imported via YAXArrays:

```julia
import YAXArrays.DD: At
```

File: src/analysis/sensitivity.jl [hunk 2]
```diff
@@ -5,7 +5,9 @@ using Logging
 using
     AxisKeys,
     NamedDims,
-    StaticArrays
+    StaticArrays,
+    YAXArrays,
+    DimensionalData
```
Comment (ConnectedSystems) [inline src/analysis/sensitivity.jl:10]: As with previous comment - if this is for `At`, import it directly.

**Comment by ConnectedSystems** [comment]
> > Does the switch to YAXArray break the ability to plot results? I don't see any compatibility layer in place.
> 
> There are also changes to `viz.rsa()` in this PR so plotting works

Thanks Rose, clearly I'm going blind.

File: ext/AvizExt/viz/sensitivity.jl [hunk 3]
```diff
@@ -175,7 +175,7 @@ Makie figure
 function ADRIA.viz.rsa!(
     g::Union{GridLayout,GridPosition},
     rs::ResultSet,
-    si::NamedDimsArray,
+    si::YAXArrays.Dataset,
```
Comment (ConnectedSystems) [inline ext/AvizExt/viz/sensitivity.jl:178]: Why not import `Dataset` directly so prefacing it with `YAXArrays` is not necessary?

Or is there a different `Dataset` type that is causing a conflict?

File: src/analysis/sensitivity.jl [hunk 4]
```diff
@@ -451,78 +455,107 @@ ADRIA.sensitivity.rsa(X, y; S=10)
...
 function rsa(
-    X::DataFrame, y::AbstractVector{<:Real}, model_spec::DataFrame; S::Int64=10
-)::NamedDimsArray
+    X::DataFrame, y::AbstractVector{<:Real}, factors::Vector{Symbol}, model_spec::DataFrame;
+    S::Int64=10,
+)::YAXArrays.Dataset
     N, D = size(X)
 
-    X_di = zeros(N)
+    X_i = zeros(N)
```
Comment (ConnectedSystems) [inline src/analysis/sensitivity.jl:463]: There was a reason for X_di ($X_{d, i}$) but that may now be lost to the sands of time...

Comment (Rosejoycrocker) [reply src/analysis/sensitivity.jl:463]: Ah makes sense. Should I revert or leave as is?

Comment (ConnectedSystems) [reply src/analysis/sensitivity.jl:463]: Leave as is

File: ext/AvizExt/viz/viz.jl (hunk 1 above)
Comment (Rosejoycrocker) [reply ext/AvizExt/viz/viz.jl]: I think I just took this from the YAXArray docs and forgot to remove, I've removed it now

File: ext/AvizExt/viz/sensitivity.jl (hunk 3 above)
Comment (Rosejoycrocker) [reply ext/AvizExt/viz/sensitivity.jl]: I've replaced with `Dataset`

**Comment by Rosejoycrocker** [comment]
> Some minor comments for consideration

I think I've addressed your comments, If you're happy with this I'll do another PR with similar changes for `outcome_map`

**Comment by ConnectedSystems** [comment]
Busy with something else at the moment, would you mind attaching a figure here to demonstrate its working please?

**Comment by Rosejoycrocker** [comment]
> Busy with something else at the moment, would you mind attaching a figure here to demonstrate its working please?

Here's an example 
sensitivity_rsa_ex

**Comment by ConnectedSystems** [comment]
The peak is telling you that there is high variability in that factor space.

In other words, the outcome assessed is highly sensitive to that region of factor space, which corresponds with no intervention.

**Comment by Rosejoycrocker** [comment]
> The peak is telling you that there is high variability in that factor space.
> 
> In other words, the outcome assessed is highly sensitive to that region of factor space, which corresponds with no intervention.

Makes sense

**Review by ConnectedSystems — APPROVED** [review]
LGTM

**Pull request merged**